lib/api/errors/internal_error.rb

```diff
@@ -34,15 +34,25 @@ class InternalError < ErrorBase
       identifier 'InternalServerError'
       code 500
 
-      def initialize(error_message = nil, **)
+      def initialize(error_message = nil, exception: nil, **)
         error = I18n.t('api_v3.errors.code_500')
 
-        if error_message
+        if error_message && !hidden_exception?(exception)
           error += " #{error_message}"
         end
 
         super error
       end
+
+      private
+
+      ##
+      # Hide internal database errors in production
+      def hidden_exception?(exception)
+        return false unless Rails.env.production?
```

Comment:
```suggestion
        Rails.env.production? && exception.is_a?(ActiveRecord::StatementInvalid)
```
I would remove the condition, as it reads better (for me) without `unless`.